frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -0,0 +1,255 @@
+import './_restore-pvc-modal.scss';
+
+import * as React from 'react';
+
+import { Form, FormGroup, Grid, GridItem, TextInput, Alert } from '@patternfly/react-core';
+import {
+  HandlePromiseProps,
+  ResourceIcon,
+  withHandlePromise,
+  validate,
+  resourceObjPath,
+  history,
+  RequestSizeInput,
+  Timestamp,
+  LoadingInline,
+} from '@console/internal/components/utils';
+import {
+  K8sResourceKind,
+  k8sCreate,
+  referenceFor,
+  VolumeSnapshotKind,
+  StorageClassResourceKind,
+  k8sGet,
+  PersistentVolumeClaimKind,
+} from '@console/internal/module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+import {
+  NamespaceModel,
+  PersistentVolumeClaimModel,
+  VolumeSnapshotModel,
+  VolumeSnapshotClassModel,
+} from '@console/internal/models';
+import { getName, getNamespace, Status } from '@console/shared';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+
+const dropdownUnits = {
+  Mi: 'MiB',
+  Gi: 'GiB',
+  Ti: 'TiB',
+};
+
+export const RestorePVCModal = withHandlePromise((props: RestorePVCModalProps) => {
+  const { close, cancel, resource, errorMessage, inProgress, handlePromise } = props;
+  const [restorePVCName, setPVCName] = React.useState(`${getName(resource) || 'pvc'}-restore`);
+  const defaultSize: string[] = resource?.status?.restoreSize
+    ? validate.split(resource?.status?.restoreSize)
+    : null;
+  const pvcRequestedSize = resource?.status?.restoreSize
+    ? `${defaultSize[0]} ${dropdownUnits[defaultSize[1]]}`
+    : '';
+  const [requestedSize, setRequestedSize] = React.useState(defaultSize ? defaultSize[0] : '');
+  const [requestedUnit, setRequestedUnit] = React.useState(defaultSize ? defaultSize[1] : 'Gi');
+  const [pvcSC, setPVCStorageClass] = React.useState<string>(null);
+  const [progress, setProgress] = React.useState(true);
+  const [driver, setDriver] = React.useState<string>(null);
+  const [pvcData, setPVCData] = React.useState<PersistentVolumeClaimKind>(null);
+  const namespace = getNamespace(resource);
+
+  React.useEffect(() => {
+    Promise.all([
+      k8sGet(
+        PersistentVolumeClaimModel,
+        resource?.spec?.source?.persistentVolumeClaimName,
+        namespace,
+      ).catch((e) => {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching objects', e); // Not important to show user
+        setProgress(false);
+      }),
+      k8sGet(VolumeSnapshotClassModel, resource.spec.volumeSnapshotClassName),
```

Comment:
better approach would be store the promises in variables, and handle the error in single catch block as same behavior is implemented.